tests/test_schema.py

```diff
@@ -184,3 +184,12 @@ def test_schema_get_column_type(self):
 
         schema = MappingSchema({"foo": {"bar": parse_one("INT", into=exp.DataType)}})
         self.assertEqual(schema.get_column_type("foo", "bar").this, exp.DataType.Type.INT)
+
+    def test_schema_normalization(self):
+        schema = ensure_schema(
+            {"x": {"y": {"z": {"a": "INT", '"B"': "VARCHAR"}, "w": {"C": "INT"}}}}
+        )
+        expected_mapping = {"x": {"y": {"z": {"a": "INT", '"B"': "VARCHAR"}, "w": {"c": "INT"}}}}
+
+        schema.normalize()
+        self.assertEqual(schema.mapping, expected_mapping)
```

Comment:
While `Schema.mapping` isn't explicitly private, this still feels like we're checking some private attribute of the schema that might change.

Can we test this using more explicitly public APIs? e.g. "column_names"?

Reply:
Good point. Building on this, should we also prefix `mapping` with an underscore to make it feel more "private"? I see that it's only used in the executor's `__init__.py` file at the moment, but maybe we could move that logic inside the `Tables` class.

Reply:
Possibly. Maybe let's worry about that later?